lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
Just as a heads up you gotta put the `cmd` first because the protocol will do dynamic dispatch off of the first argument. 

Yeah I was trying to suggest something more like that above where maybe the room executes the command and passing in a room struct. But yeah there might need to be a custom thing created so commands have everything available to them. Naively it could be `%{ player: %Character{}, room: %Room{}, actions: [] }` or something. The `:actions` key is for keeping track of everything the command triggered. If we wanted to keep commands side effect free then the functions the command uses could tack on what should happen into the `:actions` key. It pushes the side effects to the `Room` and makes each command I think unit testable.

Dummy example of what a `push button` command could trigger:
```elixir
def execute(cmd, ctxt) do
    ctxt
    |> spawn_creature(Troll)
    |> notify_player(Message.info("You have sprung a trap!")
    |> set_timeout({:second, 60}, %ResetTrapEvent{ trap: cmd.target }) 
end
```

So the resulting context object would have 3 actions in it for the room to perform.